# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../e2fsprogs/e2fsprogs-1.42-upstream-fixes-1.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From ba37bb704f4ab2631a39c32b35d6bb339e17b293 Mon Sep 17 00:00:00 2001
From: Theodore Ts'o <tytso@mit.edu>
Date: Mon, 5 Dec 2011 12:35:38 -0500
Subject: [PATCH 1/2] libext2fs: don't break when ext2fs_clear_generic_bmap() for 32-bit bitmaps

This is only an issue for programs compiled against e2fsprogs 1.41
that manipulate bitmaps directly.  Fortunately there are very few
programs which do that, especially those that try to clear a bitmap.

Addresses-Sourceforge-Bugs: #3451486

Reported-by: robi6@users.sourceforge.net
Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
---
 lib/ext2fs/gen_bitmap64.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git e2fsprogs-1.42/lib/ext2fs/gen_bitmap64.c e2fsprogs-1.42-upstream-fixes/lib/ext2fs/gen_bitmap64.c
index 9dbbf9f..4dc4e08 100644
--- e2fsprogs-1.42/lib/ext2fs/gen_bitmap64.c
+++ e2fsprogs-1.42-upstream-fixes/lib/ext2fs/gen_bitmap64.c
@@ -297,8 +297,8 @@ void ext2fs_clear_generic_bmap(ext2fs_generic_bitmap bitmap)
 {
 	if (EXT2FS_IS_32_BITMAP(bitmap))
 		ext2fs_clear_generic_bitmap(bitmap);
-
-	bitmap->bitmap_ops->clear_bmap (bitmap);
+	else
+		bitmap->bitmap_ops->clear_bmap (bitmap);
 }
 
 int ext2fs_mark_generic_bmap(ext2fs_generic_bitmap bitmap,
-- 
1.7.2.3


From 24c91184d6577271f7387962c90626c973389f00 Mon Sep 17 00:00:00 2001
From: Theodore Ts'o <tytso@mit.edu>
Date: Fri, 16 Dec 2011 15:36:40 -0500
Subject: [PATCH 2/2] e2fsck: fix use of uninitialized value in the MMP code

Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
---
 e2fsck/pass1.c  |    2 +-
 e2fsck/pass1b.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git e2fsprogs-1.42/e2fsck/pass1.c e2fsprogs-1.42-upstream-fixes/e2fsck/pass1.c
index 00e46d0..d225026 100644
--- e2fsprogs-1.42/e2fsck/pass1.c
+++ e2fsprogs-1.42-upstream-fixes/e2fsck/pass1.c
@@ -545,7 +545,7 @@ void e2fsck_pass1(e2fsck_t ctx)
 	int	i;
 	__u64	max_sizes;
 	ext2_filsys fs = ctx->fs;
-	ext2_ino_t	ino;
+	ext2_ino_t	ino = 0;
 	struct ext2_inode *inode;
 	ext2_inode_scan	scan;
 	char		*block_buf;
diff --git e2fsprogs-1.42/e2fsck/pass1b.c e2fsprogs-1.42-upstream-fixes/e2fsck/pass1b.c
index f7ce8e4..a9eecd2 100644
--- e2fsprogs-1.42/e2fsck/pass1b.c
+++ e2fsprogs-1.42-upstream-fixes/e2fsck/pass1b.c
@@ -267,7 +267,7 @@ struct process_block_struct {
 static void pass1b(e2fsck_t ctx, char *block_buf)
 {
 	ext2_filsys fs = ctx->fs;
-	ext2_ino_t ino;
+	ext2_ino_t ino = 0;
 	struct ext2_inode inode;
 	ext2_inode_scan	scan;
 	struct process_block_struct pb;
-- 
1.7.2.3

